packages/updatePrescriptionStatus/src/updatePrescriptionStatus.ts

```diff
@@ -2,194 +2,157 @@
 import {Logger} from "@aws-lambda-powertools/logger"
 import {injectLambdaContext} from "@aws-lambda-powertools/logger/middleware"
-import {DynamoDBClient, PutItemCommand} from "@aws-sdk/client-dynamodb"
-import {marshall} from "@aws-sdk/util-dynamodb"
 import middy from "@middy/core"
 import inputOutputLogger from "@middy/input-output-logger"
 import errorHandler from "@nhs/fhir-middy-error-handler"
+import {Bundle, BundleEntry, Task} from "fhir/r4"
+
+import {transactionBundle, validateTask} from "./validation/content"
+import {
+  accepted,
+  badRequest,
+  bundleWrap,
+  createSuccessResponseEntries,
+  serverError
+} from "./utils/responses"
+import {persistDataItems} from "./utils/databaseClient"
 
 const logger = new Logger({serviceName: "updatePrescriptionStatus"})
-const client = new DynamoDBClient({region: "eu-west-2"})
-const tableName = process.env.TABLE_NAME
-
-interface DynamoDBItem {
-  RequestID: string | undefined;
-  PrescriptionID: string;
-  PatientNHSNumber: string;
-  PharmacyODSCode: string;
-  TaskID: string;
-  LineItemID: string;
-  TerminalStatus: string;
-  RequestMessage: any;
+
+export interface DataItem {
+  LastModified: string
+  LineItemID: string
+  PatientNHSNumber: string
+  PharmacyODSCode: string
+  PrescriptionID: string
+  RequestID: string
+  RequestMessage: Task
+  Status: string
+  TaskID: string
+  TerminalStatus: string
 }
 
 const lambdaHandler = async (event: APIGatewayProxyEvent): Promise<APIGatewayProxyResult> => {
-  const xRequestId = event.headers["x-request-id"]
-  let requestBody
-  try {
-    requestBody = JSON.parse(event.body || "")
-  } catch (jsonParseError) {
-    logger.error("Error parsing JSON", {error: jsonParseError})
-    const errorResponseBody = {
-      resourceType: "OperationOutcome",
-      issue: [
-        {
-          code: "value",
-          severity: "error",
-          details: {
-            coding: [
-              {
-                system: "https://fhir.nhs.uk/CodeSystem/http-error-codes",
-                code: "BAD_REQUEST",
-                display: "400: The Server was unable to process the request."
-              }
-            ]
-          }
-        }
-      ]
-    }
-    return {
-      statusCode: 400,
-      body: JSON.stringify(errorResponseBody),
-      headers: {
-        "Content-Type": "application/fhir+json",
-        "Cache-Control": "no-cache"
-      }
-    }
+  let responseEntries: Array<BundleEntry> = []
+
+  const xRequestID = getXRequestID(event, responseEntries)
+  if (!xRequestID) {
+    return response(400, responseEntries)
   }
 
-  const responseBundle: any = {
-    resourceType: "Bundle",
-    type: "transaction-response",
-    entry: []
+  const requestBody = event.body
+  const requestBundle = castEventBody(requestBody, responseEntries)
+  if(!requestBundle) {
+    return response(400, responseEntries)
   }
+  logger.info("Request audit log", {requestBody: requestBundle})
```

Comment:
we dont need this as the request is logged using middy inputOutputLogger

Reply:
Removed.